### DISCUSSION TITLE
Validation problems

### DISCUSSION THREAD

**Discussion by sonicsmooth** [AvaloniaUI/Avalonia#10270] [General]
Using 11 preview.

I'm in the codebehind for a `TemplatedControl`.
I have a direct Avalonia property `ValueProperty` with a custom type.
I've read https://docs.avaloniaui.net/docs/data-binding/data-validation and the example https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/MVVM/ValidationSample and also the Star Ratings example.

1. When I throw an exception in the property setter, it just crashes the program.  Nothing catches the exception. Do exceptions in setters for validation only work from inside a `ViewModelBase` ?
2. When I try overriding `UpdateDataValidation(AvaloniaProperty property, BindingValueType state, Exception? error)` (this is from preview 11, so it's different from the override in the stars example), it never gets called.  I have `enableDataValidation:true` in my property, and I have my main control wrapped in a `<DataValidationErrors>` tag in the xaml.  It still feels like I'm missing some connection.  I was not able to make much sense of the official documentation.

I'm also happy just setting a class value someplace and updating the UI through class styles, but I need to dig into that.  Would prefer to use the official validation mechanism.  --> (Edit: I was able to set and remove a custom `Classes` on the `TextBox` inside mycontrol after finding it using `e.Namescope.Find<...>(...)` inside `override ApplyTemplate`, and had a matching <style...> in my xaml to color the background red on failure)

thank you.

**Comment by timunie** [comment]
Can you file a simple sample App for us to test? I'd like to see it on my own. 

Thank you.
